Box.V2/Request/HttpRequestHandler.cs

```diff
@@ -18,63 +18,80 @@ public class HttpRequestHandler : IRequestHandler
             // Need to account for special cases when the return type is a stream
             bool isStream = typeof(T) == typeof(Stream);
 
-            HttpRequestMessage httpRequest = request.GetType() == typeof(BoxMultiPartRequest) ?
-                                                BuildMultiPartRequest(request as BoxMultiPartRequest) :
-                                                BuildRequest(request);
-
-            // Add headers
-            foreach (var kvp in request.HttpHeaders)
-                httpRequest.Headers.TryAddWithoutValidation(kvp.Key, kvp.Value);
-
-            // If we are retrieving a stream, we should return without reading the entire response
-            HttpCompletionOption completionOption = isStream ?
-                HttpCompletionOption.ResponseHeadersRead :
-                HttpCompletionOption.ResponseContentRead;
-
-            Debug.WriteLine(string.Format("RequestUri: {0}", httpRequest.RequestUri));//, RequestHeader: {1} , httpRequest.Headers.Select(i => string.Format("{0}:{1}", i.Key, i.Value)).Aggregate((i, j) => i + "," + j)));
-
             try
             {
-                HttpClient client = CreateClient(request);
-
-                HttpResponseMessage response = await client.SendAsync(httpRequest, completionOption).ConfigureAwait(false);
-
-                BoxResponse<T> boxResponse = new BoxResponse<T>();
-                boxResponse.Headers = response.Headers;
+                int attempts = 0;
+                BoxResponse<T> boxResponse = null;
 
-                // Translate the status codes that interest us 
-                boxResponse.StatusCode = response.StatusCode;
-                switch (response.StatusCode)
+                while (attempts < 10) // For now only 10 retries
```

Comment:
I agree... for now I'll put it as a constant, but it makes sense to make it configurable